src/map/map.cpp

```diff
@@ -1870,7 +1870,7 @@ int map_addflooritem(struct item *item, int amount, int16 m, int16 x, int16 y, i
 
 	if (!map_searchrandfreecell(m,&x,&y,flags&2?1:0))
 		return 0;
-	r = rnd();
+	r = rnd(UINT32_MAX);
```

Comment:
I am too lazy too figure out the math below.
But it should be changed as well and additionally it should also be moved down.